openxr/examples/vulkan.rs

```diff
@@ -772,5 +1091,101 @@ struct Framebuffer {
     color: vk::ImageView,
 }
 
-/// Maximum number of frames in flight
-const PIPELINE_DEPTH: u32 = 2;
+#[repr(C)]
+#[derive(Clone, Copy)]
+struct Vertex {
+    pos: [f32; 3],
+    color: [f32; 3],
+}
+
+impl Vertex {
+    pub fn cube(color: [f32; 3]) -> impl Iterator<Item = Self> {
+        [1, 0, 3, 2, 6, 0, 4, 1, 5, 3, 7, 6, 5, 4]
+            .into_iter()
+            .map(move |i| Self {
+                pos: [(i & 1) as f32, ((i >> 1) & 1) as f32, ((i >> 2) & 1) as f32],
+                color: color.clone(),
+            })
+    }
```

Comment:
It might be a lot more concise to hardcode the geometry into the vertex shader (indexed using `gl_VertexIndex`) rather than using a conventional vertex buffer that we hardcode data into anyway.

Reply:
I agree it would be more concise to hardcode the vertices (and how I'm generating these was inspired by what the vertex shader was previously doing), but I think for the example it may make sense to show something closer to the average use case where a user will want to upload their models into a vertex buffer.